### PR TITLE
Li hong yue patch 3

### PR DESCRIPTION
**Pull request by li-hong-yue**
I tried many things, including but not limited to:

qk norm,
weight tying,
muon optimizer,
softcap logits,
u-net shaped transformers,
zero init classification
linformer,

etc

but none of these worked!!!
only the original transformer architecture works ?!
why? I am so sad :> 

Nevermind, I guess I am just unlucky in the era of ML alchemy.
Thank you for your effort to create this assignment!

### CODE REVIEW

**Comment by li-hong-yue** [comment]
Sorry, but my result is not merged to the leaderboard.
Thank you!
